### PR TITLE
v0.6 changelog

### PR DESCRIPTION
**Pull request by choldgraf**
This PR includes the changelog for v0.6. It also updates the documentation around making releases in general.

### CODE REVIEW

**Comment by choldgraf** [comment]
@yuvipanda could you take a look at the steps outlined in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/460/files#diff-6a3371457528722a734f3c51d9238c13R57 ? I'm sure I'm missing some things!

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,164 @@
...
+
+This release is primarily focused on better support
+for Autoscaling, Microsoft Azure support & better
+default security. There are also a number of bug fixes
+and configurability improvements!
+
+### Breaking changes
+
+#### Pre-puller configuration
+In prior versions, if you wanted to disable the pre-puller,
```
Comment (willingc) [inline CHANGELOG.md]: Let's put in ...versions (0.5 or lower),

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,164 @@
...
+you would use:
+
+```yaml
+prePuller:
+   enabled: false
+```
+
+Now, to get the same effect, you need to use:
```
Comment (willingc) [inline CHANGELOG.md]: Now, to disable the pre-puller,...

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,164 @@
...
+
+```yaml
+prePuller:
+   hook:
+     enabled: false
+```
+
+See the pre-puller docs for more info!
```
Comment (willingc) [inline CHANGELOG.md]: Link?

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,164 @@
...
+
+### Upgrading from 0.5
+
+This release does not require any special steps to upgrade from v0.4. See the [upgrade documentation](https://zero-to-jupyterhub.readthedocs.io/en/latest/upgrading.html)
```
Comment (willingc) [inline CHANGELOG.md]: Wrong version at end of sentence.

File: CHANGELOG.md [hunk 5]
```diff
@@ -6,6 +6,164 @@
...
+cluster is being used. In this release, we made a few
+changes to let z2jh interact better with the autoscaler!
+
+- Configure z2jh to 'pack' your users onto nodes, rather
+  than 'spread' them across nodes. ([link](http://zero-to-jupyterhub.readthedocs.io/en/latest/advanced.html#picking-a-scheduler-strategy))
+- A 'continuous' pre-puller that allows user images to
+  be pulled on new nodes easily, leading to faster startup
+  times for users on new nodes. ([link](http://zero-to-jupyterhub.readthedocs.io/en/latest/advanced.html?highlight=prepull#pre-pulling-images-for-faster-startup))
+- Hub and Proxy pod will not be disrupted by autoscaler,
+  by using [PodDisruptionBudget](https://kubernetes.io/docs/tasks/run-application/configure-pdb/)s. They will also stick
```
Comment (willingc) [inline CHANGELOG.md]: ~~They~~ The Hub and Proxy

File: CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,164 @@
...
+#### Better configurability
+
+We now have better documentation and bug fixes for configurability!
+
+- `extraConfig` can be a dictionary instead of just a
+  string. This helps when you have to split your `config.yaml`
+  into multiple files for complex deployments
+- How user storage works by default is better documented! ([link](http://zero-to-jupyterhub.readthedocs.io/en/latest/user-storage.html))
+- Reading config in `extraConfig` from `extraConfigMap` now actually works!
+- You can configure the URL users are spawned into. This allows defaulting users to JupyterLab. ([link](http://zero-to-jupyterhub.readthedocs.io/en/latest/user-environment.html#use-jupyterlab-by-default))
```
Comment (willingc) [inline CHANGELOG.md]: You can configure the URL that the spawner uses for its users.

**Comment by yuvipanda** [comment]
I think I addressed all your comments, @willingc!

I think this is ready for review now. I'll take a pass at the upgrade instructions in a different PR.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -52,3 +52,21 @@ development.
...
+   entries in the [CHANGELOG](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CHANGELOG.md).
+2. Double-check that there aren't any un-documented breaking changes.
+3. If applicable, include a section in the [CHANGELOG](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/CHANGELOG.md)
+   that contains upgrade instructions (usually only applicable for major changes to the code).
+4. Generate a list of contributors since the latest release. Use the script
+  in `tools/contributors.py` to list all contributions (anyone who made a
+  commit or a comment)    since the latest release. Update the dates in that
+  script, run it, and paste the output into the changelog. For an example,
+  see [the v0.5 list of contributors](https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/v0.5/CHANGELOG.md#contributors).
+5. ...
```
Comment (choldgraf) [inline CONTRIBUTING.md]: just a note that we should update this whenever we go through the release process

Comment (yuvipanda) [reply CONTRIBUTING.md]: whoops I didn't even notice that! I'm writing it up in https://hackmd.io/BwJghgxgjAnCAsBaYAGeBmR8CmB2GyAbIRIjAEbooAmKEMAZrmOUA===#

**Pull request merged**